internal/internal.go

```diff
@@ -391,5 +394,39 @@ func parseTime(format string, timestamp string, location *time.Location) (time.T
 	case "stampnano":
 		format = time.StampNano
 	}
-	return time.ParseInLocation(format, timestamp, loc)
+
+	if !strings.Contains(format, "MST") {
+		return time.ParseInLocation(format, timestamp, loc)
+	}
+
+	// Golang does not parse times with ambiguous timezone abbreviations,
+	// but only parses the time-fields and the timezone NAME with a zero
+	// offset (see https://groups.google.com/g/golang-nuts/c/hDMdnm_jUFQ/m/yeL9IHOsAQAJ).
+	// To handle those timezones correctly we can use the timezone-name and
+	// force parsing the time in that timezone. This way we get the correct
+	// time for the "most probably" of the ambiguous timezone-abbreviations.
+	ts, err := time.Parse(format, timestamp)
+	if err != nil {
+		return time.Time{}, err
+	}
+	zone, offset := ts.Zone()
+	if zone == "UTC" || offset != 0 {
+		return ts.In(loc), nil
+	}
+	once.Do(func() {
+		log.Print("W! You are using abbreviated timezones for which parsing was fixed in v1.27.0.")
+		log.Print("W! Please remove any workarounds in place and check your resulting times carefully!")
+		log.Print("W! In case you experience any problems, please file an issue!")
```

Comment:
```suggestion
		log.Print("W! If you experience any problems, please file an issue on GitHub")
```

Reply:
IMHO this line can be removed, this counts for any problem experienced, not only for timezones.

Reply:
@Hipska I like to encourage people to report the stuff, so the line is well spend IMO. This worked quite well for the [gnmi plugin](https://github.com/influxdata/telegraf/blob/9c6cd94d6e46bba8ce1971685f17417b2409bd90/plugins/inputs/gnmi/gnmi.go#L33)...

Reply:
Okay, but there it is in a single log message, not spread over multiple independent lines.. Please keep the messages together!